### PR TITLE
cuda: enable checkpoint support for paused tasks

### PR DESCRIPTION
**Pull request by rst0git**
If a CUDA process is already in a "locked" or "checkpointed" state during `criu dump`, the CUDA plugin fails with an error because it attempts an unnecessary "lock" action using the `cuda-checkpoint` tool. This pull request extends the CUDA plugin to handle such case by first checking the original state of the CUDA process and skipping unnecessary "lock" and "checkpoint" actions if the process was already locked or checkpointed before CRIU was invoked.

### CODE REVIEW

File: plugins/cuda/cuda_plugin.c [hunk 1]
```diff
@@ -397,7 +460,7 @@ int cuda_plugin_pause_devices(int pid)
 		return -1;
 	}
 
-	if (add_pid_to_buf(&cuda_pids, pid)) {
+	if (add_pid_to_buf(&cuda_pids, restore_tid, CUDA_TASK_RUNNING)) {
```
Comment (jesus-ramos) [inline plugins/cuda/cuda_plugin.c:463]: Shouldn't we be tracking the PID and not the restore thread id here?

**Review by jesus-ramos — APPROVED** [review]
LGTM

File: plugins/cuda/cuda_plugin.c (hunk 1 above)
Comment (rst0git) [reply plugins/cuda/cuda_plugin.c]: You're right, tracking the PID would indeed be more reliable here. Thank you for catching that!

**Comment by rst0git** [comment]
> Is there any real use-case for that?

The use-case is similar to https://github.com/checkpoint-restore/criu/pull/2514 -- the CUDA tasks may be in a "locked" or "checkpointed" state before `criu dump` is invoked to ensure consistent checkpoint/restore, particularly in distributed model training where multiple containers are running across different cluster nodes.

**Comment by rst0git** [comment]
@avagin I've updated the commit message with a brief description of the use-cases for this functionality.

**Pull request merged**